pkgs/tools/typesetting/tex/texlive/default.nix

```diff
@@ -186,8 +184,19 @@ let
       );
 
   # combine a set of TL packages into a single TL meta-package
-  combinePkgs = pkgList: lib.concatLists # uniqueness is handled in `combine`
-    (builtins.map (a: a.pkgs) pkgList);
+  combinePkgs = pkgList: lib.catAttrs "pkg" (
+    let
+      # a TeX package is an attribute set { pkgs = [ ... ]; ... } where pkgs is a list of derivations
+      # the derivations make up the TeX package and optionally (for backward compatibility) its dependencies
+      tlPkgToSets = { pkgs, ... }: builtins.map (pkg: {
+          key = "${pkg.pname}.${pkg.tlType or ""}-${pkg.version}-${pkg.outputName or "notDrv"}";
+          inherit pkg;
+        }) pkgs;
+      pkgListToSets = lib.concatMap tlPkgToSets; in
+    builtins.genericClosure {
```

Comment:
If keys are unique after this, do we still need https://github.com/NixOS/nixpkgs/blob/2dfe04b2d2373d552d782ac677b9348fd3bf66e4/pkgs/tools/typesetting/tex/texlive/combine.nix#L36 ?

Reply:
Correct, we don't, provided we also add `hasFormats`, `hasHyphens` only to the "run" variant of each package. Incidentally, this is the correct thing to do, otherwise importing the docs without the texmf files (if you ever want to do that!) would make `texlive.combine` think that certain formats must be generated, causing the build to fail.

Reply:
(Resolved in additional commit that removes `sortedUniqueStrings` and is more careful with the `has*` attributes.)